docs/getting_started/java/first_program_in_java/index.md

```diff
@@ -1,373 +1,757 @@
+If you do not supply one, the Service creates a random unique ID to use instead.
+
+### The Transfer application
+
+The "money transfer" app is a small Java program that creates a Workflow Execution.
+In a real deployments, you might invoke similar code when someone submits a form, presses a button, or visits a certain URL.
+Here is the code that performs this work:
+
+<!--SNIPSTART money-transfer-java-initiate-transfer {"selectedLines": ["31-70"]}-->
 <!--SNIPEND-->
 
-Run the `TransferMoneyWorker` class from IntelliJ, or run the following command from the project root in separate terminal:
+This code builds a Temporal Client, which is a type that can communicate with the Temporal Service.
+The Client is used to create a new Workflow stub that points to the Workflow method.
+With the details for the transfer collected together in a single instance (`transaction`), the `WorkflowClient` type creates a new Workflow Execution using the stub and the `transaction` instance.
+The execution runs asynchronously.
+After printing out details, the application exits.
+
+The Workflow Execution (Event History and other metadata) lives on the Temporal Service.
+Exiting the application does not affect this.
+
+### Run the money transfer app
+
+To run the app, issue this Maven command:
+
+```
+mvn compile exec:java \
+    -Dexec.mainClass="moneytransferapp.TransferApp" \
+    -Dorg.slf4j.simpleLogger.defaultLogLevel=warn
+```
+
+It creates a new transfer request for a random amount between $15 and $75:
+
+
+```bash
+MONEY TRANSFER PROJECT
+
+Initiating transfer of $45 from [Account 711713707] to [Account 221056126].
 
-```command
-./gradlew startWorker
+[WorkflowID: money-transfer-workflow]
+[RunID: 37688cca-ffa2-48cf-809b-f18f5119bca3]
+[Transaction Reference: a0061bb5-a718-47cb]
 ```
 
-When you start the Worker it begins polling the Task Queue.
-The first Task the Worker finds is the one that tells it to execute the Workflow function.
-The Worker communicates the event back to the server which then causes the server to send Activity Tasks to the Task Queue as well.
-The Worker then grabs each of the Activity Tasks in their respective order from the Task Queue and executes each of the corresponding Activities.
-You will get a console output showing that both activity tasks were executed by the Worker:
+The request is produced as a Workflow Execution and sent to the Temporal Server.
+The Temporal Server is an essential part of the overall system, but requires additional components for operation.
+The complete system is known as the Temporal Service, which is a deployment of the Temporal Server, plus the additional components used with it such as a database like Apache Cassandra, PostgreSQL, or MySQL.
+
+:::note
+
+This app uses a hard-coded "money-transfer-workflow" string for the Workflow ID.
+If you run the transfer app a second time, you'll raise an error unless a Worker is running and has already processed the previous request.
```

Comment:
My memory is foggy on this, but is this true? I know it's true in Python but I thought Java just ignored it if you tried to invoke a Workflow with the same workflow-id as one already running.

Reply:
Just tested this. It's not true in Java. Java will just fail silently.

Reply:
I had errors, but I'm going to omit because it's really not that important.